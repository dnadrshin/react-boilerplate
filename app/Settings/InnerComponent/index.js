//@flow
import React from 'react'

const
	InnerComponent = (props: {
		increase?: () => {}
	}) => <div>
		<button onClick={props.increase}>Async get-set</button>
	</div>

export default InnerComponent
