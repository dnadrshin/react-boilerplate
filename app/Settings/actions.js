import keyMirror from 'keymirror';

const
	dataSet = id => ({id, type: 'SET'}),

	types = keyMirror({
		DECREASE: null,
		INCREASE: null,
		SET     : null,
		GET     : null,
	})

export default {dataSet, types}
