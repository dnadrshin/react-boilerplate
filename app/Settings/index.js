import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import actions from './actions'

class Settings extends React.Component {
	constructor(props) {
		super(props)
	}

	doSet(id) {
		this.props.set(10)
	}

	render(){
		return <div>
			<h1>Settings {this.props.state.settings.counter}</h1>
			<button onClick={this.props.increase}>+</button>
			<button onClick={this.props.decrease}>-</button>
			<button onClick={() => this.doSet()}>set</button>
		</div>
	}
}

export default connect(
	state => ({
		state: state
	}),

	(dispatch, props) => ({
		set     : id => dispatch(actions.dataSet(id)),
		increase: () => dispatch({type:'INCREASE'}),
		decrease: () => dispatch({type:'DECREASE'})
	})
)(Settings)
