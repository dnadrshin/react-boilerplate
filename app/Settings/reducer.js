import {createReducer} from '../helpers'
import actions from './actions'

const
	initialState = {counter: 5};

export default (state = initialState, action = {}) => createReducer(state, action, {
	[actions.types.SET]: () => ({
		...state,
		counter: action.id,
	}),

	[actions.types.INCREASE]: () => ({
		...state,
		counter: state.counter+1,
	}),

	[actions.types.DECREASE]: () => ({
		...state,
		counter: state.counter-1,
	}),
});
