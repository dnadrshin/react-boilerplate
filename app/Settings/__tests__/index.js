import Settings from '../';
import configureStore from 'redux-mock-store';
import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

const
	instance = mount(<Settings store={configureStore()({
		settings: {
			
		}
	})} />);

describe('Settings has 4 buttons', () => {
	it('has footer bar shown across all pages', () => {
		expect(toJson(instance)).toMatchSnapshot();
		expect(instance.find('button')).toHaveLength(4);
	});
});
