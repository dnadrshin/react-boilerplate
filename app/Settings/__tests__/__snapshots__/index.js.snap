// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings has 4 buttons has footer bar shown across all pages 1`] = `
<Connect(Settings)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Settings
    decrease={[Function]}
    get={[Function]}
    increase={[Function]}
    set={[Function]}
    settings={Object {}}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <h1>
        Settings 
      </h1>
      <button
        onClick={[Function]}
      >
        +
      </button>
      <button
        onClick={[Function]}
      >
        -
      </button>
      <button
        onClick={[Function]}
      >
        set
      </button>
      <InnerComponent
        increase={[Function]}
      >
        <div>
          <button
            onClick={[Function]}
          >
            Async get-set
          </button>
        </div>
      </InnerComponent>
    </div>
  </Settings>
</Connect(Settings)>
`;
